examples/gcoap/gcoap_cli.c

```diff
@@ -40,15 +42,39 @@ static const coap_resource_t _resources[] = {
     { "/riot/board", COAP_GET, _riot_board_handler, NULL },
 };
 
+static const char *_link_params[] = {
+    ";ct=0;rt=\"count\"",
+    NULL
+};
+
 static gcoap_listener_t _listener = {
     &_resources[0],
     sizeof(_resources) / sizeof(_resources[0]),
+    _encode_link,
     NULL
 };
 
 /* Counts requests sent by CLI. */
 static uint16_t req_count = 0;
 
+/* Adds link format params to resource list */
+static ssize_t _encode_link(const coap_resource_t *resource, char *buf,
+                            size_t maxlen, coap_link_encoder_ctx_t *context) {
+    ssize_t res = gcoap_encode_link(resource, buf, maxlen, context);
+    if (res > 0) {
+        if (_link_params[context->link_pos]
+                && (strlen(_link_params[context->link_pos]) < (maxlen - res))) {
```

Comment:
I agree, not worth it. I see this code used only infrequently. In addition, my vision is that we will use a blockwise mechanism to write most/all payloads. This will localize the buffer boundary test code to one place, which helps a lot with the bloat of these kinds of tests all over the place.